src/Shared/ProjectWriter.cs

```diff
@@ -118,8 +118,8 @@ internal void Initialize(XmlDocument project, XmlDeclaration projectRootElementD
             }
 
             // don't write an XML declaration unless the project already has one or has non-default encoding
-            _writeXmlDeclaration = (projectRootElementDeclaration != null) ||
-                                   (_documentEncoding != null && !Equals(_documentEncoding, Encoding.UTF8));
+            _writeXmlDeclaration = projectRootElementDeclaration != null ||
```

Comment:
No, if there was an xml declaration in the file we always write it (`true`). If there wasn't we only write it if the encoding is not UTF8.